File: app/src/main/java/com/sakurafish/pockettoushituryou/view/helper/AdsHelper.java
package com.sakurafish.pockettoushituryou.view.helper;

import android.content.Context;

import com.google.android.gms.ads.AdRequest;
import com.sakurafish.pockettoushituryou.R;
import com.sakurafish.pockettoushituryou.pref.Pref;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
public class AdsHelper {

    private static final String TAG = AdsHelper.class.getSimpleName();
    // Intent action filter string
    public static final String ACTION_BANNER_CLICK = "ACTION_BANNER_CLICK";
    // Intent extras key string
    public static final String INTENT_EXTRAS_KEY_CLASS = "INTENT_EXTRAS_KEY_CLASS";
    // click interval time millisecond
    public static final long CLICK_DELAY_MILLIS = 1000 * 60 * 60; // 60 min

    private final Context context;
    private final Pref pref;

    @Inject
    public AdsHelper(Context context, Pref pref) {
        this.context = context;
        this.pref = pref;
    }

    public boolean isIntervalOK() {
        // Need 60 min Interval
        long now = System.currentTimeMillis();
        long lastClickTime = getLastClickTimeMillis();
        if (lastClickTime == 0) return true;
        long passedTime = now - lastClickTime;
        return passedTime >= CLICK_DELAY_MILLIS;
    }

    public AdRequest getAdRequest() {
        String[] ids = context.getResources().getStringArray(R.array.admob_test_device);
        AdRequest.Builder builder = new AdRequest.Builder();
        for (String id : ids) {
            builder.addTestDevice(id);
        }
        return builder.build();
    }

    public long getLastClickTimeMillis() {
        return pref.getPrefLong(context.getString(R.string.PREF_LAST_AD_CLICK_TIME), 0);
    }

    public void setLastClickTimeMillis(final long date) {
        pref.setPrefLong(context.getString(R.string.PREF_LAST_AD_CLICK_TIME), date);
    }

    public long getIntervalTimeMillis(final long now, final long lastClickTimeMillis) {
        long intervalTimeMillis = CLICK_DELAY_MILLIS;
        if (lastClickTimeMillis != 0) {
            long passedTimeMillis = now - lastClickTimeMillis;
            intervalTimeMillis = CLICK_DELAY_MILLIS - passedTimeMillis;
            if (intervalTimeMillis > CLICK_DELAY_MILLIS) {
                intervalTimeMillis = CLICK_DELAY_MILLIS;
            }
        }
        return intervalTimeMillis;
    }
}

File: app/src/main/java/com/sakurafish/pockettoushituryou/view/helper/AdsHelper.kt
package com.sakurafish.pockettoushituryou.view.helper

import android.content.Context

import com.google.android.gms.ads.AdRequest
import com.sakurafish.pockettoushituryou.R
import com.sakurafish.pockettoushituryou.shared.Pref

import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class AdsHelper @Inject constructor(private val context: Context, private val pref: Pref) {

    // Need 5 minutes Interval
    val isIntervalOK: Boolean
        get() {
            val now = System.currentTimeMillis()
            val lastClickTime = lastClickTimeMillis
            if (lastClickTime == 0L) return true
            val passedTime = now - lastClickTime
            return passedTime >= CLICK_DELAY_MILLIS
        }

    val adRequest: AdRequest
        get() {
            val ids = context.resources.getStringArray(R.array.admob_test_device)
            val builder = AdRequest.Builder()
            for (id in ids) {
                builder.addTestDevice(id)
            }
            return builder.build()
        }

    var lastClickTimeMillis: Long
        get() = pref.getPrefLong(context.getString(R.string.PREF_LAST_AD_CLICK_TIME), 0)
        set(date) = pref.setPrefLong(context.getString(R.string.PREF_LAST_AD_CLICK_TIME), date)

    fun getIntervalTimeMillis(now: Long, lastClickTimeMillis: Long): Long {
        var intervalTimeMillis = CLICK_DELAY_MILLIS
        if (lastClickTimeMillis != 0L) {
            val passedTimeMillis = now - lastClickTimeMillis
            intervalTimeMillis = CLICK_DELAY_MILLIS - passedTimeMillis
            if (intervalTimeMillis > CLICK_DELAY_MILLIS) {
                intervalTimeMillis = CLICK_DELAY_MILLIS
            }
        }
        return intervalTimeMillis
    }

    companion object {
        // Intent action filter string
        const val ACTION_BANNER_CLICK = "ACTION_BANNER_CLICK"
        // Intent extras key string
        const val INTENT_EXTRAS_KEY_CLASS = "INTENT_EXTRAS_KEY_CLASS"
        // click interval time millisecond
        const val CLICK_DELAY_MILLIS = (1000 * 60 * 5).toLong() // 5 min
    }
}
